app/Helpers/database/achievement.php

```diff
@@ -226,67 +221,84 @@ function UploadNewAchievement(
             return false;
         }
 
-        $query = "
-            INSERT INTO Achievements (
-                ID, GameID, Title, Description,
-                MemAddr, Progress, ProgressMax,
-                ProgressFormat, Points, Flags, type,
-                Author, DateCreated, DateModified,
-                Updated, VotesPos, VotesNeg,
-                BadgeName, DisplayOrder, AssocVideo,
-                TrueRatio
-            )
-            VALUES (
-                NULL, '$gameID', '$title', '$desc',
-                '$mem', '$progress', '$progressMax',
-                '$progressFmt', $points, $flag, $typeValue,
-                '$dbAuthor', NOW(), NOW(),
-                NOW(), 0, 0,
-                '$badge', 0, NULL,
-                0
-            )";
-        $db = getMysqliConnection();
-        if (mysqli_query($db, $query) !== false) {
-            $idInOut = mysqli_insert_id($db);
-            postActivity($author, ActivityType::UploadAchievement, $idInOut);
-
-            static_addnewachievement($idInOut);
-            addArticleComment(
-                "Server",
-                ArticleType::Achievement,
-                $idInOut,
-                "$author uploaded this achievement.",
-                $author
-            );
-
-            // uploaded new achievement
-            AchievementCreated::dispatch(Achievement::find($idInOut));
-
-            return true;
-        }
+        $achievement = new Achievement();
+        $achievement->GameID = $gameID;
+        $achievement->Title = $title;
+        $achievement->Description = $desc;
+        $achievement->MemAddr = $mem;
+        $achievement->Points = $points;
+        $achievement->Flags = $flag;
+        $achievement->type = ($typeValue == 'NULL') ? null : $type;
+        $achievement->Author = $author;
+        $achievement->BadgeName = $badge;
+
+        $achievement->save();
+        $idInOut = $achievement->ID;
+        postActivity($author, ActivityType::UploadAchievement, $idInOut);
+
+        static_addnewachievement($idInOut);
+        addArticleComment(
+            "Server",
+            ArticleType::Achievement,
+            $idInOut,
+            "$author uploaded this achievement.",
+            $author
+        );
+
+        // uploaded new achievement
+        AchievementCreated::dispatch($achievement);
```

Comment:
Would it make sense to dispatch AchievementPublished if the flag matches? This way it would already cascade into the games metrics update.
Not sure if we actually need AchievementCreated for that.

Reply:
nevermind - makes sense for `achievements_unpublished` as you mentioned

Reply:
Achievements created in the DLL are always uploaded to Unofficial, then promoted from there. There's no way (currently) to upload directly to Core. As such, `AchievementPublished` seems incorrect.

Reply:
perfect, thanks for the clarification!